lib/stripe/stripe_object.rb

```diff
@@ -72,13 +72,17 @@ def refresh_from(values, opts, partial=false)
     #
     # * +values+ - Hash of values to use to update the current attributes of
     #   the object.
+    # * +:opts+ Options for StripeObject like an API key.
     #
     # ==== Options
     #
-    # * +:opts+ Options for StripeObject like an API key.
-    def update_attributes(values, opts = {})
+    # * +:dirty+ Whether values should be initiated as "dirty" (unsaved).
+    #   Defaults to true.
+    def update_attributes(values, opts = nil, options = {})
+      dirty = options[:dirty]
       values.each do |k, v|
-        @values[k] = Util.convert_to_stripe_object(v, opts)
+        @values[k] = Util.convert_to_stripe_object(v, opts || {})
+        dirty_value!(@values[k]) unless dirty == false
```

Comment:
If I'm understanding this correctly, `dirty` defaults to "true" by getting the value `nil` in `dirty = options[:dirty]`, which gives a false result for `dirty == false`, and then the `unless` causes us to make `dirty_value!` call. That seems pretty tricky to me, in particular the difference between `nil` and `false` for the `:dirty` option. I'm not sure I have a better idea though, just thought I'd flag it.